lib/sailor.js

```diff
@@ -386,38 +387,39 @@ class Sailor {
                                 OBJECT_STORAGE_SIZE_THRESHOLD: settings.OBJECT_STORAGE_SIZE_THRESHOLD
                             }
                         );
-                        const [bodyId, ...passthroughIds] = await Promise.all([
-                            that.uploadMessageBody(bodyBuf),
-                            ...passthroughBufs.map(async ({ stepId, body, id }) => {
-                                const bodyId = id || await that.uploadMessageBody(body);
-                                return { stepId, bodyId };
-                            })
-                        ]);
-
-                        if (bodyId) {
-                            logger.info('Message body uploaded', { id: bodyId });
-                            const { headers } = data;
-                            data.body = {};
-                            data.headers = {
-                                ...(headers || {}),
-                                [OBJECT_ID_HEADER]: bodyId
-                            };
-                        } else {
+
+                        let bodyId;
+                        let passthroughIds;
+                        try {
+                            [bodyId, ...passthroughIds] = await Promise.all([
+                                that.uploadMessageBody(bodyBuf),
+                                ...passthroughBufs.map(async ({ stepId, body, id }) => {
+                                    const bodyId = id || await that.uploadMessageBody(body);
+                                    return { stepId, bodyId };
+                                })
+                            ]);
+                        } catch (e) {
                             logger.info('Message body not uploaded');
+                            e.message = `Maester error: ${e.toString()}`;
+                            return onError(e);
```

Comment:
```suggestion
                            return onError(new Error("Lightweight message/passthrough body upload error"));
```
This info will be enough for customer to see in errors. And additional details will be visible in logs.